packages/xml-views-validation/src/validators/attributes/use-of-depracated-attribute.ts

```diff
@@ -0,0 +1,78 @@
+import { find } from "lodash";
+import { XMLAttribute } from "@xml-tools/ast";
+import {
+  UI5SemanticModel,
+  UI5Prop,
+  UI5Event,
+  UI5Association,
+} from "@ui5-language-assistant/semantic-model-types";
+import {
+  flattenEvents,
+  getUI5ClassByXMLElement,
+  flattenAssociations,
+  flattenProperties,
+} from "@ui5-language-assistant/logic-utils";
+import { UseOfDeprecatedAttributeIssue } from "../../../api";
+import {
+  buildDeprecatedIssueMessage,
+  DeprecatedUI5Symbol,
+} from "../../utils/deprecated-message-builder";
+
+export function validateUseOfDeprecatedAttribute(
+  attribute: XMLAttribute,
+  model: UI5SemanticModel
+): UseOfDeprecatedAttributeIssue[] {
+  if (attribute.syntax.key === undefined || attribute.key === null) {
+    // Can't give an error without a position or value
+    return [];
+  }
+
+  const ui5PropEventAssociation = findUI5PropEventAssociationByXMLAttributeKey(
+    attribute,
+    model
+  );
+
+  if (
+    ui5PropEventAssociation === undefined ||
+    ui5PropEventAssociation.deprecatedInfo === undefined
+  ) {
+    return [];
+  }
+
+  const deprecatedIssue: UseOfDeprecatedAttributeIssue = {
+    kind: "UseOfDeprecatedAttribute",
```

Comment:
if we will support multiple more specific kinds of issue here.
maybe we should extract the switchcase from `buildDeprecatedIssueMessage`
something like:

```typescript
function getDeprecatedDetails(symbol: UI5DeprecatedSymbol): 
{
  issueKind:"UseOfDeprecatedEvent" | "UseOfDeperecatedProperty" /* ... */,
  symbolName: string,
  symbolKindName: "Property" | "Event" | "Association" /*...*/
}
```